pkg/api/validation/validation_test.go

```diff
@@ -724,6 +724,10 @@ func TestValidatePodSpec(t *testing.T) {
 			RestartPolicy: api.RestartPolicyAlways,
```

Comment:
All other *failure* cases here need to have at least one container. Otherwise, they aren't actually catching the failure that they expect. Sometimes it's helpful to check the length of the errors, but that also doesn't cover all the cases here.